### PR TITLE
test: Add Integration Tests

### LINKED ISSUE #15: Add Integration Tests

### PR DESCRIPTION
**Pull request by fqutishat**
closes #15

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: .ci-properties [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+if [ $IS_RELEASE == false ]
+then
+  EXTRA_VERSION=snapshot-$(git rev-parse --short=7 HEAD)
+  PROJECT_VERSION=$BASE_VERSION-$EXTRA_VERSION
+else
+  PROJECT_VERSION=$BASE_VERSION
+fi
+
+export SIDETREE_FABRIC_NODE_TAG=$ARCH-$PROJECT_VERSION
```
Comment (troyronda) [inline .ci-properties]: There should be no references to Fabric in this project.

File: test/bddtests/fixtures/.env [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright SecureKey Technologies Inc. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# This file contains the default images and tags used in the docker-based
+# Fabric fixtures. The images and tags can be overridden using environment
```
Comment (troyronda) [inline test/bddtests/fixtures/.env]: There should be no references to Fabric in this project.

File: .ci-properties (hunk 1 above)
Comment (fqutishat) [reply .ci-properties]: done

File: test/bddtests/fixtures/.env (hunk 2 above)
Comment (fqutishat) [reply test/bddtests/fixtures/.env]: done

File: test/bddtests/compose.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	return containerIDs, err
+}
+
+func (c *Composition) refreshContainerList() (err error) {
+	var allAPIContainers []docker.APIContainers
+	var thisProjectsContainers []docker.APIContainers
+	if thisProjectsContainers, err = c.dockerClient.ListContainers(docker.ListContainersOptions{All: true, Filters: map[string][]string{"name": {c.projectName}}}); err != nil {
+		return fmt.Errorf("Error refreshing container list for project '%s':  %s", c.projectName, err)
+	}
+	//if allApiContainers, err = c.dockerClient.ListContainers(docker.ListContainersOptions{All: true}); err != nil {
```
Comment (troyronda) [inline test/bddtests/compose.go]: leftover code.

File: test/bddtests/restclient/sender.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+	"github.com/trustbloc/sidetree-node/models"
+)
+
+type HttpRespone struct {
+	Payload  []byte
+	ErrorMsg string
+}
+
+// SendRequest sends a regular POST request to the sidetree-kevlar
```
Comment (sandrask) [inline test/bddtests/restclient/sender.go]: remove kevlar references

File: test/bddtests/restclient/sender.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	}
+
+	httpReq, err := http.NewRequest("POST", url, bytes.NewReader(b))
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(httpReq)
```
Comment (sandrask) [inline test/bddtests/restclient/sender.go]: just return client.Do() here

File: test/bddtests/restclient/sender.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+		return nil, err
+	}
+	return resp, nil
+}
+func decode(response *http.Response, v interface{}) error {
+	respBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
+	}
+	if err := json.NewDecoder(strings.NewReader(string(respBytes))).Decode(v); err != nil {
```
Comment (sandrask) [inline test/bddtests/restclient/sender.go]: return json.NewDecoder()

File: test/bddtests/did_sidetree_steps.go [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+	data, _ := ioutil.ReadAll(r)
+	d := strings.Replace(string(data), "#DIDID", didID, -1)
+	d = strings.Replace(d, "#CONTEXT", generateUUID(), -1)
+
+	return base64.URLEncoding.EncodeToString([]byte(d))
+}
+
+// RegisterSteps registers did sidetree steps
+func (d *DIDSideSteps) RegisterSteps(s *godog.Suite) {
+	s.Step(`^client send request to create DID document "([^"]*)" as "([^"]*)" with DID id "([^"]*)"$`, d.sendDIDDocumentWithID)
```
Comment (sandrask) [inline test/bddtests/did_sidetree_steps.go]: client sends request

File: test/bddtests/did_sidetree_steps.go [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+	s.Step(`^check error response contain "([^"]*)"$`, d.checkErrorResp)
+	s.Step(`^client send request to create DID document "([^"]*)" as "([^"]*)"`, d.sendDIDDocument)
+	s.Step(`^check success response contain "([^"]*)"$`, d.checkSuccessResp)
```
Comment (sandrask) [inline test/bddtests/did_sidetree_steps.go]: response contains

File: test/bddtests/fixtures/config/didDocumentTemplate.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
+
+{
+  "@context": "https://w3id.org/did/v1",
+  #DIDID
```
Comment (sandrask) [inline test/bddtests/fixtures/config/didDocumentTemplate.json]: we should have proper document without DIDID as template, DID ID can be added dynamically for negative tests

File: test/bddtests/restclient/sender.go (hunk 4 above)
Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: done

File: test/bddtests/compose.go (hunk 3 above)
Comment (fqutishat) [reply test/bddtests/compose.go]: done

File: test/bddtests/restclient/sender.go (hunk 5 above)
Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: done

File: test/bddtests/restclient/sender.go (hunk 6 above)
Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: done

File: test/bddtests/did_sidetree_steps.go (hunk 7 above)
Comment (fqutishat) [reply test/bddtests/did_sidetree_steps.go]: done

File: test/bddtests/did_sidetree_steps.go (hunk 8 above)
Comment (fqutishat) [reply test/bddtests/did_sidetree_steps.go]: done

File: test/bddtests/fixtures/config/didDocumentTemplate.json (hunk 9 above)
Comment (fqutishat) [reply test/bddtests/fixtures/config/didDocumentTemplate.json]: done

**Pull request merged**